src/dodal/beamlines/i18.py

```diff
@@ -0,0 +1,218 @@
+@skip_device()
+def slits_1(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> Slits:
+    return numbered_slits(
+        1,
+        # BL18I-AL-SLITS-01
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+# Must document what PandAs are physically connected to
+# See: https://github.com/bluesky/ophyd-async/issues/284
+@skip_device()
+def panda1(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> HDFPanda:
+    return device_instantiation(
+        HDFPanda,
+        "panda1",
+        "-MO-PANDA-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        directory_provider=get_directory_provider(),
+    )
+
+
+@skip_device()
+def xspress3(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> Xspress3:
+    """
+    16 channels Xspress3 detector
+        -EA-XPS-02:CAM:MaxSizeX_RBV
+      also ArraySize
+      also :CONNECTED
+    """
+
+    return device_instantiation(
+        Xspress3,
+        # prefix="-EA-DET-03:",
+        prefix="-EA-XPS-02:",
+        name="Xspress3",
+        num_channels=16,
+        wait=wait_for_connection,
+        fake=fake_with_ophyd_sim,
+    )
+
+
+def dcm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> DoubleCrystalMonochromator:
+    return device_instantiation(
+        DoubleCrystalMonochromator,
+        "dcm",
+        "",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        bl_prefix=False,
+        motion_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-MO-DCM-01:",
+        temperature_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-DI-DCM-01:",
+        crystal_1_metadata=CrystalMetadata(
+            usage="Bragg",
+            type="silicon",
+            reflection=(1, 1, 1),
+            d_spacing=(3.13475, "nm"),
+        ),
+        crystal_2_metadata=CrystalMetadata(
+            usage="Bragg",
+            type="silicon",
+            reflection=(1, 1, 1),
+            d_spacing=(3.13475, "nm"),
+        ),
+    )
+
+
+@skip_device()
+def i0(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "i0",
+        "-EA-XBPM-02:",
+        # -DI-XBPM-02:DEV:Firmware
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="Cividec Diamond XBPM",
+        directory_provider=get_directory_provider(),
+    )
+
+
+@skip_device()
+def it(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "it",
+        "-EA-TTRM-02:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="PIN Diode",
+        directory_provider=get_directory_provider(),
+    )
+
+
+def vfm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> FocusingMirror:
```

Comment:
I18 have controls for their KB mirror (vfm and hfm) :
   X, Y, PITCH, BEND1, BEND2, CURVE, ELLIP
Whereas FocusingMirror has : 
   X, Y, X1, X2, Y1, Y2, Y3, YAW, PITCH, ROLL
So, many PVs will fail to connect and we should use a different class for this. Probably need to make an I18 specific one. But that might not be necessary- since individual motors are controlled during the mirror focus optimisation scans.

Reply:
covered in the new issue, skipping device for now